### ISSUE TITLE
Unable to run simple service

### ISSUE THREAD

**Issue by nieznanysprawiciel** [golemfactory/yagna#2171]
Name: nieznanysprawiciel
yagna version:  0.11.0-r5
OS+lang+version (if applicable): py, python 3.8.10
Title: Simple service unable to start

```jsx
nieznanysprawiciel@nieznanysprawiciel-laptop-Leopard-9SCXR:~/Repos/test-session/yapapi/examples/simple-service-poc$ python simple_service.py --show-usage --subnet-tag hybrid --running-time 35000
[2022-08-16T12:32:56.358+0200 INFO yapapi] Using log file simple-service-yapapi-2022-08-16_12.32.56.log; in case of errors look for additional information there
[2022-08-16T12:32:57.585+0200 INFO yapapi.summary] [Job 1] Agreement proposed to provider 'etam.h' (0x25c5f09f11a06f8324ef4d892011372cf97c45ee)
[2022-08-16T12:32:57.638+0200 INFO yapapi.summary] [Job 1] Agreement confirmed by provider 'etam.h'
[2022-08-16T12:32:59.166+0200 INFO yapapi.services.service_runner] <SimpleService starting on etam.h [ 0x25c5f09f11a06f8324ef4d892011372cf97c45ee ]> commissioned
[2022-08-16T12:32:59.359+0200 INFO yapapi.summary] Received proposals from 106 providers so far
[2022-08-16T12:33:08.363+0200 INFO yapapi.summary] Received proposals from 111 providers so far
[2022-08-16T12:35:08.414+0200 INFO yapapi.summary] Received proposals from 115 providers so fa
```

Interesting log from `yagna`

```jsx
[2022-08-16T12:33:43.355+0200 DEBUG ya_service_bus::typed] Call to GetAgreement failed: Not found: c0c7cdb32b54f3a8647a3dffc41a27dfaae3e5064bb6e99400d761b8c1c5ea85
```

Logs:
https://www.notion.so/golemnetwork/requestor-hybrid-net-testing-session-fa30d680c2814354a1b4490dd76d56df#aff91772178c45c5a8cae60bb283fdaa

https://www.notion.so/golemnetwork/requestor-hybrid-net-testing-session-fa30d680c2814354a1b4490dd76d56df#6d4fbe13c6444bf0bfad47c234603ea5

https://www.notion.so/golemnetwork/requestor-hybrid-net-testing-session-fa30d680c2814354a1b4490dd76d56df#fa37f377b45d47efb638db4fc0582129

**Comment by wkargul** [comment]
As per `core/activity/src/api.rs` anytime when either call `get_activity_usage_web` or `get_activity_state_web` identity is verified. We need to know if deamon is running as Provider or Requestor. Therefore GSB call is made underneath. Naturally one call is ending with NotFound error which is logged in debug. Please do not consider the issue from above as the bug.

What is interesting if deamon is running as Provider and Requestor paralelly, the current implementation does not cover this scenario. It returns prematurely.

To sum it up: something else caused the issue with running `python simple_service.py --show-usage --subnet-tag hybrid --running-time 35000` by @nieznanysprawiciel . I have prepared another issue for research on the root cause of `simple service` beeing unable to start -> https://github.com/golemfactory/yagna/issues/2185
